src/blight/actions/bitcode_extract.py

```diff
@@ -0,0 +1,40 @@
+"""
+The `BitcodeExtract` action.
+"""
+
+import logging
+import os
+import subprocess
+from typing import List
+
+from blight.action import CompilerAction
+from blight.enums import Lang
+from blight.tool import CompilerTool
+
+logger = logging.getLogger(__name__)
+
+
+class BitcodeExtract(CompilerAction):
+    """
+    Compile and extract bitcode. The output bitcode file will be located in the same direcory
+    as the output executable.
+    """
+
+    def before_run(self, tool: CompilerTool) -> None:  # type: ignore
+        if tool.lang in [Lang.C, Lang.Cxx]:
+            args: List[str]
+            if tool.outputs:
+                args = ["-c", "-emit-llvm", "-o", tool.outputs[0] + ".bc", *tool.inputs]
+            else:
+                logger.debug(
+                    "not extracting bitcode with unspecified output location"
+                )  # pragma: no cover
+                assert False  # pragma: no cover
+
+            bitcode_flags = os.getenv("LLVM_BITCODE_GENERATION_FLAGS")
```

Comment:
Just curiosity: why are you getting these from the environment, instead of baking them into the action?

Reply:
I was mirroring this: https://github.com/SRI-CSL/gllvm#customizing-the-bitcode-generation-eg-lto. I thought it might make it the easier to replace but I can bake it into the action.

Reply:
I didn't even know that existed! Cool. Okay, that's reasonable then, although let's use the `config` mechanism to plumb this instead of a separate environment variable. In other words, we should do something like this:

```bash
BLIGHT_ACTION_BITCODEEXTRACT="store=/tmp/whatever llvm-flags='-flto'"
```

...which will make `self._config` into `{"store": "/tmp/whatever", "llvm-flags": "-flto"}`.